packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -304,6 +304,11 @@ class Hero extends StatefulWidget {
             inviteHero(hero, tag);
           }
         }
+      } else if (element.widget is Offstage) {
```

Comment:
Special-casing the Offstage widget is not a good idea as there may be other widgets that will make a Hero "unreachable".  We should see if we can fix this problem in a more generic way.

Reply:
Agree, It's not a good design that Hero knows Offstage. However, the current implementation throws exceptions, I think it is a reasonable first aid.

Reply:
I agree with @goderbauer. This is actually a violation of the Flutter style guide: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes.
Furthermore, addressing the root issue is preferable to a 'first aid' or temporary solution. What is the root issue here that is causing the exception?

Reply:
Root issue: When there are multiple Navigators such as CupertinoPageScaffold, Hero searches for an inactive Navigators that is not related to the Hero-animating being performed. If the same Heroes exists in different Navigators, the active Navigator's Hero should be selected.

Reply:
Should each Navigators have a widget like HeroStage that indicates whether or not to participate in a Hero animation and should be switched according to active / inactive? The mechanism is similar to Offstage, but it is a way to explicitly add Hero-related widgets.